examples/kafka/cluster-2x-ext-zk-svc-descriptor.yaml

```diff
@@ -0,0 +1,86 @@
+apiVersion: apps.kubeblocks.io/v1
+kind: Cluster
+metadata:
+  name: kafka2-ext-zk-descriptor
+  namespace: default
+spec:
+  terminationPolicy: Delete
+  # Specifies the name of the ClusterDefinition to use when creating a Cluster.
+  # Note: DO NOT UPDATE THIS FIELD
+  # The value must be `kafaka` to create a Kafka Cluster
+  clusterDef: kafka
+  # Specifies the name of the ClusterTopology to be used when creating the
+  # Cluster.
+  # - combined: combined Kafka controller (KRaft) and broker in one Component
+  # - combined_monitor: combined mode with monitor component
+  # - separated: separated KRaft and Broker Components.
+  # - separated_monitor: separated mode with monitor component
+  # Valid options are: [combined,combined_monitor,separated,separated_monitor,kafka2-external-zk]
+  topology: kafka2-external-zk
+  services:
+    - name: bootstrap
+      serviceName: bootstrap
+      componentSelector: kafka-broker
+      spec:
+        type: ClusterIP
+        ports:
+          - name: kafka-client
+            targetPort: 9092
+            port: 9092
+  componentSpecs:
+    - name: kafka-broker
+      componentDef: kafka27-broker
+      serviceRefs:
+        - name: kafkaZookeeper
+          namespace: default
+          serviceDescriptor: kafka-cluster-zookeeper-service
+      tls: false
+      replicas: 3
+      serviceAccountName: kb-kafka-cluster
+      env:
+        - name: KB_BROKER_DIRECT_POD_ACCESS
+          value: 'true'
+        - name: KAFKA_HEAP_OPTS
+          value: "-Xmx500m -Xms500m"
+      resources:
+        limits:
+          cpu: '0.5'
+          memory: 0.5Gi
+        requests:
+          cpu: '0.5'
+          memory: 0.5Gi
+      volumeClaimTemplates:
+        - name: data
+          spec:
+            accessModes:
+              - ReadWriteOnce
+            resources:
+              requests:
+                storage: 20Gi
+        - name: metadata
+          spec:
+            storageClassName: null
+            accessModes:
+              - ReadWriteOnce
+            resources:
+              requests:
+                storage: 5Gi
+---
+apiVersion: apps.kubeblocks.io/v1
+kind: ServiceDescriptor
+metadata:
+  name: kafka-cluster-zookeeper-service
+  namespace: default
+spec:
+  # Specifies the type or nature of the service.
+  # Should represent a well-known application cluster type, such as {mysql, redis, zookeeper}.
+  serviceKind: zookeeper
+  # Represents the version of the service reference.
+  serviceVersion: 3.8.5
+  # Represents the endpoint of the service connection credential.
+  endpoint:
+    # your external zk endpoints here
+    value: "zookeeper-cluster-zookeeper-0.zookeeper-cluster-zookeeper-headless.default:2181,zookeeper-cluster-zookeeper-1.zookeeper-cluster-zookeeper-headless.default:2181,zookeeper-cluster-zookeeper-2.zookeeper-cluster-zookeeper-headless.default:2181/test"
```

Comment:
LGTM. However, for scenarios where multiple clusters share zk, can we consider entrusting the control logic of metadata conflicts to addon instead of relying on specifying '/test' when defining ServiceDescriptor?

For example:
- the only thing defined in ServiceDescriptor is the zk base address:
```
 endpoint:
    # your external zk endpoints here
    value: "zk-1.zookeeper-cluster-zookeeper-headless.default:2181"
```
- an env defined in cmpd-broker such as: ZK_SUB_PATH=kafka1, the default value is KB_CLUSTER_NAME. Finally, in broker-env.tpl, KB_KAFKA_ZOOKEEPER_CONN will process into 'zk-1.zookeeper-cluster-zookeeper-headless.default:2181/kafka1'

The benefits of this in shared zk are:
- Avoid defining a ServiceDescriptor cr for a set of kafka, but define one for each zk
- The core configuration for controlling zk path conflicts is in the Kafka Cluster yaml, not the weakly controlled object ServiceDescriptor

Reply:
Good advice，using env KB_KAFKA_ZK_SUB_PATH